cmd/containerd-shim/main_unix.go

```diff
@@ -59,6 +60,9 @@ func init() {
 	flag.StringVar(&runtimeRootFlag, "runtime-root", proc.RuncRoot, "root directory for the runtime")
 	flag.StringVar(&criuFlag, "criu", "", "path to criu binary")
 	flag.BoolVar(&systemdCgroupFlag, "systemd-cgroup", false, "set runtime to use systemd-cgroup")
+	// currently, the `containerd publish` utility is embedded in the daemon binary for some reason.
```

Comment:
Maybe something more like this:

> The default publisher is embedded in `containerd`, user wanted to change it should adhere to the same CLI and set this variable to their binary path otherwise `os.Executable()` is used.

Reply:
`containers-shim` itself does not use os.Executable

Reply:
I see. Just get rid of `for some reason` then

Reply:
done